azext_iot/common/_azure.py

```diff
@@ -9,6 +9,25 @@
 from azext_iot.common.utility import validate_key_value_pairs
 from azext_iot.common.utility import trim_from_start
 from azext_iot._factory import iot_hub_service_factory
+from azure.cli.core._profile import Profile
+
```

Comment:
nit: If you are so inclined, please rename azext_iot/common/_azure.py to azext_iot/common/azure.py (no underscore). The modules usage evolved overtime doesn't need a underscore now. If not, we'll do it  in a subsequent PR.

Reply:
I tried to do this, but I ran into issues with it conflicting with the Azure module, it may be best to do this separately :)